File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/preset/InventoryPresetSaveResultMessage.scala





















// Generated on 05/03/2013 10:34:58
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait InventoryPresetSaveResultMessage extends MessageDeserializer {
  val messageId = 6170

  protected trait Message extends BaseMessage with Serializable {
    val messageId = InventoryPresetSaveResultMessage.messageId

    val presetId: Byte
    val code: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(presetId)
      buf.writeByte(code)
    }

  }

  protected def create(presetId: Byte, code: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val presetId = buf.readByte()
    val code = buf.readByte()

    create(presetId, code)
  }

}

object InventoryPresetSaveResultMessage extends InventoryPresetSaveResultMessage {
  case class MessageImpl(presetId: Byte, code: Byte) extends Message
  type Target = MessageImpl

  override def create(presetId: Byte, code: Byte) = MessageImpl(presetId, code)

  def apply(presetId: Byte, code: Byte) = create(presetId, code)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/InventoryPresetSaveResultMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InventoryPresetSaveResultMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6170;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte presetId;
    public byte code;
    
    public InventoryPresetSaveResultMessage() { }
    
    public InventoryPresetSaveResultMessage(byte presetId, byte code) {
        this.presetId = presetId;
        this.code = code;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(presetId);
        buf.writeByte(code);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        presetId = buf.readByte();
        if (presetId < 0)
            throw new RuntimeException("Forbidden value on presetId = " + presetId + ", it doesn't respect the following condition : presetId < 0");
        code = buf.readByte();
        if (code < 0)
            throw new RuntimeException("Forbidden value on code = " + code + ", it doesn't respect the following condition : code < 0");
    }
    
}
